vllm/entrypoints/openai/serving_score.py

```diff
@@ -62,134 +44,300 @@ def __init__(
                          models=models,
                          request_logger=request_logger)
 
-    async def create_score(
+    async def _embedding_score(
         self,
-        request: ScoreRequest,
-        raw_request: Optional[Request] = None,
-    ) -> Union[ScoreResponse, ErrorResponse]:
-        """
-        Score API similar to Sentence Transformers cross encoder
+        tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+        text_1: List[str],
+        text_2: List[str],
+        request: Union[RerankRequest, ScoreRequest],
+        request_id=str,
+        tokenization_kwargs: Optional[dict[str, Any]] = None,
+        lora_request: Optional[Union[LoRARequest, None]] = None,
+        prompt_adapter_request: Optional[Union[PromptAdapterRequest,
+                                               None]] = None,
+        trace_headers: Optional[Mapping[str, str]] = None,
+    ) -> List[PoolingRequestOutput]:
+
+        input_texts = text_1 + text_2
+
+        engine_prompts: List[TokensPrompt] = []
+        tokenize_async = make_async(tokenizer.__call__,
+                                    executor=self._tokenizer_executor)
+
+        tokenization_kwargs = tokenization_kwargs or {}
+        tokenized_prompts = [
+            tokenize_async(t, **tokenization_kwargs) for t in input_texts
+        ]
+
+        for tok_result, input_text in zip(
+                await asyncio.gather(*tokenized_prompts), input_texts):
```

Comment:
```suggestion
        tokenized_prompts = await asyncio.gather(*(
            tokenize_async(t, **tokenization_kwargs) for t in input_texts
        ))

        for tok_result, input_text in zip(tokenized_prompts, input_texts):
```

I think this is a bit more readable as the async stuff becomes closer together.